### PR TITLE
Refactor smart comment parsing

### PR DESCRIPTION
**Pull request by diesal11**
I've been doing some digging into Postgraphile V5 and trying to build my own Plugin. While learning the codebase i decided to tackle a smaller issue i found with smart comment parsing. The following SQL:

```sql
COMMENT ON TABLE reports_public.report_base IS $$
    @name ReportBase

    @interface mode:relational type: report_type
    @type Foo references: foo
    @type Bar references: bar
$$;
```

Is currently parsed by the `parseSmartComment` as:

```json
{
    "name": "ReportBase",
    "description": "@interface mode:relational type: report_type\n@type Foo...."
}
```

The issue here is the `@interface` & `@type` tags being parsed as part of the description, rather than their own tags. This may be intentional, but i found it unintuitive and didn't realise that the extra `\n` i had placed between `@name` & `@interface` was causing none of the tags to be picked up by the Polymorphism plugin later. I think it's worth supporting extraneous empty lines by simply ignoring them.

This PR refactors the code so that:
* All empty lines are ignored
* Any line not starting with `@` is included in the documentation.
* It also adds some unit test cases.
* Some changeset stuff the GH but was complaining about, sorry if i did that wrong.

With these changes, the following SQL:

```sql
COMMENT ON TABLE test_table IS $$
This is a description.

@tag1 value1
@tag2 value2
How many of these can i add?
@tag3

This is also a description.

This is another description, with a newline before and after it.

$$;
```
```ts
{
      tags: {
        tag1: "value1",
        tag2: "value2",
        tag3: true,
      },
      description:
        "This is a description.\n\nHow many of these can i add?\nThis is also a description.\n\nThis is another description, with a newline before and after it.",
    }
```

I know this wasn't pre-discussed so please feel free to close if you don't think this change is a good fit or needs further changes.

### CODE REVIEW

**Comment by benjie** [comment]
The problem with this is it means there’s no way to start a comment with an `@`; for example `@benjie indicated CITEXT is appropriate here`. Currently the double newline is used as a “the description follows” break.

**Comment by diesal11** [comment]
Ah, I knew i would be overlooking something....

I could update it to support breaking an `@` with a `\` on a given line. I.E. `\@benjie indicated CITEXT is appropriate here` ?

But that would also be a breaking change. The simpler solution may just be to be explicit the effect of a double newline into the docs?

**Comment by benjie** [comment]
Yes; please feel free to add that as part of the smart comments specification if it's not already clear. There should be an "edit this page" link on the relevant page of the docs.

I'd also be open to a PR that output a warning for comments that start with an `@`, though there would need to be a way to explicitly mark it as deliberate (e.g. `\@`; but if you add that then you also need to add `\\`... And it would change the format for existing comments (though this is unlikely to have been hit). And then are you escaping everything or just the first character? Also putting `\` inside an `E` string in Postgres is not ideal:

```
comment on thing is
  E'@omit create\n@name Whatever\n\n\\@benjie says this is fine';
```
)

So yeah... Probably docs only for now.

File: postgraphile/website/postgraphile/smart-comments.md [hunk 1]
```diff
@@ -81,6 +81,38 @@ This field has a load of arbitrary tags.
 $$;
 ```
 
+Note that adding two newlines in a row will be interpreted as the start of a multi-line comment, no tags added after the first
+newline will be parsed & will be treated as part of the description.
```
Comment (benjie) [inline postgraphile/website/postgraphile/smart-comments.md:85]:
```suggestion
Note that adding two newlines in a row separates the smart tags section from the
descriptive prose. Content after two newlines will not be parsed for smart tags
even if it starts with an `@` character.
```

**Review by benjie — CHANGES_REQUESTED** [review]
Awesome; thanks!

**Review by benjie — APPROVED** [review]

File: postgraphile/website/postgraphile/smart-comments.md [hunk 2]
```diff
@@ -81,6 +81,39 @@ This field has a load of arbitrary tags.
 $$;
 ```
 
+Note that adding two newlines in a row separates the smart tags section from the
+descriptive prose. Content after two newlines will not be parsed for smart tags
+even if it starts with an `@` character.
+
```
Comment (benjie) [inline postgraphile/website/postgraphile/smart-comments.md:87]:
```suggestion
even if it starts with an `@` character.

For example:

```

File: postgraphile/website/postgraphile/smart-comments.md [hunk 3]
```diff
@@ -81,6 +81,39 @@ This field has a load of arbitrary tags.
...
+@isImportant
+@jsonField date timestamp
+@jsonField name text
+
+This field has a load of arbitrary tags.
+@jsonField episode enum ONE=1 TWO=2
+$$;
+```
+
+Results in the following JSON tags object:
```
Comment (benjie) [inline postgraphile/website/postgraphile/smart-comments.md:100]:
```suggestion
results in the following JSON tags object:
```

**Review by benjie — CHANGES_REQUESTED** [review]

**Review by benjie — APPROVED** [review]

**Pull request merged**